remark-custom-rules/title-capitalization/remarkLintTitleCapitalization.js

```diff
@@ -0,0 +1,75 @@
+/*
+Rule for checking that titles have the correct word's capitalization
+(a/the is small, noun is starting from big latter etc.).
+
+Useful articles:
+https://kindlepreneur.com/title-capitalization/
+https://www.letpub.com/author_education_Title_Capitalization_and_You
+*/
+
+import { lintRule } from 'unified-lint-rule'
+import { visit } from 'unist-util-visit'
+import { capitalizeWord, isUpperCase } from './lib/utils.js'
+import lowerCaseWords from './lib/lowerCaseWords.js'
+
+const cache = {}
+
+export function checkTitle(title, exceptionWords = {}) {
+  const correctTitle = title
+    .split(' ')
+    .map((word, index) => {
+      // Checking that word is not in exception dictionary and not in uppercase.
+      if (!exceptionWords[word] && !isUpperCase(word)) {
+        word = word.toLowerCase()
+
+        // Handling a hyphenated word. For example: 'Flight-or-Fight'
+        if (word.includes('-')) {
+          word = word
+            .split('-')
+            .map(part => (lowerCaseWords[part] ? part : capitalizeWord(part)))
+            .join('-')
+
+          // Checking that word shouldn't be lowercase or if it is the first word in the title.
+        } else if (!lowerCaseWords[word] || index === 0) {
+          word = capitalizeWord(word)
+        }
+      }
+
+      return word
+    })
+    .join(' ')
```

Comment:
Not all spaces are equal :).

I often use ` ` (non-break space) to prohibit breaking line in some part of the sentence.

You can process words with saving spaces by:

```js
// Even Flight-or-Fight case is processed there
title.replace(/[^\s-]\w+/, word => {
  
})
```

Reply:
Cool way to handle strings! The code has become much cleaner :)